### PR TITLE
rfc: physical machine auth

### PR DESCRIPTION
**Pull request by iguoyr**
Signed-off-by: SiyuChen <ryougi201@gmail.com>

### CODE REVIEW

File: text/2021-09-09-physical-machine-auth.md [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+To secure the service-to-service communication, establish an mTLS connection
+between chaos-controller-manager and chaosd.
+
+The following certificates are required:
+
+- CA certificate: generated when deploying `Chaos Mesh` with `helm` or
+`instal.sh`, and saved in the secret named
+`chaos-mesh-controller-manager-client-certs` (and on each physical machine)
+- certificate of chaos-controller-manager side: generated when deploying
+`Chaos Mesh` with `helm` or `instal.sh`, and saved in the secret named
```
Comment (WangXiangUSTC) [inline text/2021-09-09-physical-machine-auth.md:99]:
```suggestion
`Chaos Mesh` with `helm` or `install.sh`, and saved in the secret named
```

File: text/2021-09-09-physical-machine-auth.md [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+cause duplicate injection problems and  it requires the chaosd service
+API to support idempotency.
+
+## Alternatives
+
+NA
+
+## Unresolved questions
+
+### Not support a role to access multiple specified `PhysicalMachines`
```
Comment (WangXiangUSTC) [inline text/2021-09-09-physical-machine-auth.md:180]: how about creating the related CRD of `PhysicalMachine` under the same namespace?

File: text/2021-09-09-physical-machine-auth.md [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+kind: PhysicalMachine
+metadata:
+  name: pm-172.16.112.130
+  namespace: chaos-testing
+  labels:
+    chaos-mesh/physical-machine-group: abc
+    kubernetes.io/arch: arm64
+    kubernetes.io/os: linux
+spec:
+  address: https://172.16.112.130:31767
```
Comment (WangXiangUSTC) [inline text/2021-09-09-physical-machine-auth.md:47]: Do we need to add the information of TLS in the crd?

File: text/2021-09-09-physical-machine-auth.md (hunk 2 above)
Comment (iguoyr) [reply text/2021-09-09-physical-machine-auth.md:180]: The problem here is that this solution still relies on the Kubernetes RBAC implementation, so the definition of roles is still only two resources, `Role` and `ClusterRole`. 
1. Using `Role`, the user only has the access of one namespace, and with global considerations, can only experiment with pods in that namespace, this may not be the result we want.
2. Using `ClusterRole`, the user has cross-namespace access, so it is not possible to restrict. 

In short, I think this is due to the mismatch between the Kubernetes RBAC mechanism and this requirement. Kubernetes only controls access at the namespace level, and it is not suitable for more granular control of resource access. And most importantly, we don't have a clear understanding of the actual user's access control needs for now, so it's hard to determine which solution is more appropriate.

File: text/2021-09-09-physical-machine-auth.md (hunk 3 above)
Comment (iguoyr) [reply text/2021-09-09-physical-machine-auth.md:47]: Do you mean to split the `address` field into `protocol`, `ip` and `port`, which is similar to the `service` or `endpoint` resource?

File: text/2021-09-09-physical-machine-auth.md (hunk 1 above)
Comment (iguoyr) [reply text/2021-09-09-physical-machine-auth.md]: done.

File: text/2021-09-09-physical-machine-auth.md (hunk 3 above)
Comment (WangXiangUSTC) [reply text/2021-09-09-physical-machine-auth.md:47]: I mean add information like  public key in the crd. I got it, it is generated by Chaos Mesh, so don't need to set it here

Comment (WangXiangUSTC) [reply text/2021-09-09-physical-machine-auth.md:47]: I mean some information like the public key. I got it, it is generated by Chaos Mesh, so don't need to set it here.

**Comment by WangXiangUSTC** [comment]
LGTM

**Review by WangXiangUSTC — APPROVED** [review]

**Review by cwen0 — APPROVED** [review]
LGTM
